modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+		return _sourceItems;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public boolean isDisabledClearAll() {
+		return _disabledClearAll;
+	}
+
+	public boolean isValid() {
+		return _valid;
+	}
+
+	public void setClearAllTitle(String clearAllTitle) {
+		_clearAllTitle = clearAllTitle;
+	}
+
+	public void setCloseButtonAriaLabel(String closeButtonAriaLabel) {
+		_closeButtonAriaLabel = closeButtonAriaLabel;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setDisabledClearAll(boolean disabledClearAll) {
+		_disabledClearAll = disabledClearAll;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setExtractData(Object extractData) {
+	}
+
+	public void setHelpText(String helpText) {
+		_helpText = helpText;
+	}
+
+	public void setInputName(String inputName) {
+		_inputName = inputName;
+	}
+
+	public void setInputValue(String inputValue) {
+		_inputValue = inputValue;
+	}
+
+	public void setIsValid(boolean valid) {
+		_valid = valid;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
+	 *             #setLocator(Object)}
+	 */
+	@Deprecated
+	public void setLabelLocator() {
+	}
+
+	public void setLocator(Object locator) {
+		_locator = locator;
+	}
+
+	public void setSelectedItems(List<Map<String, String>> selectedItems) {
+		_selectedItems = selectedItems;
+	}
+
+	public void setSourceItems(List<Map<String, String>> sourceItems) {
+		_sourceItems = sourceItems;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
+	 *             #setLocator(Object)}
+	 */
+	@Deprecated
+	public void setValueLocator() {
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_clearAllTitle = "";
+		_closeButtonAriaLabel = "";
+		_disabled = false;
+		_disabledClearAll = false;
+		_helpText = "";
+		_inputName = "";
+		_inputValue = "";
+		_label = "";
+		_locator = null;
+		_selectedItems = new ArrayList<>();
+		_sourceItems = new ArrayList<>();
+		_valid = true;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
+		props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
+		props.put("disabled", _disabled);
+		props.put("disabledClearAll", _disabledClearAll);
+		props.put("helpText", _helpText);
+		props.put("inputName", _inputName);
+		props.put("inputValue", _inputValue);
+		props.put("isValid", _valid);
+		props.put("label", _label);
+
+		if (_locator != null) {
+			props.put("locator", _locator);
+		}
+
+		props.put("selectedItems", _selectedItems);
+		props.put("sourceItems", _sourceItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
```

Comment:
Remember that here we're following the same logic as in `CheckboxTag`.  We want the `cssClass` attribute passed to the tag to be rendered as classes of the `input`, not of the main element.

So here we don't want to add `form-group` class but `form-control-inset`.

We also need to override the `writeCssClassAttribute` method that will render the main element classes, in this case `form-group`.

Then in the `processStartTag` method we need to call `super.writeCssClassAttribute` and `super.writeDynamicAttributes` where we're writing the `input`, just like we're already doing with `writeIdAttribute`.

Reply:
Thanks for that call which claryfied that too. 
I'll push a commit with these changes in a bit.